### PR TITLE
Add support for overriding the detected Kubernetes version

### LINKED ISSUE #733: CRD's will not deploy correctly with Kustomize
**Describe what happened:**
Datadogs CustomResourceDefinitions will not deploy correctly, as they are right now, with Kubernetes versions >1.21-0, when deployed using Kustomize.
The reason is that Kustomize uses `helm template`, which in turn (unless ran with --validate) does not fetch capabilities.

The root cause can be tracked here:
https://github.com/kubernetes-sigs/kustomize/issues/3816

I will submit a PR that adds the option to override the detected Kubernetes version (since the CRD's make use of capabilities) to deal with this. It also breaks the use of deployment via Anthos Config Management.

**Describe what you expected:**
Datadog should deploy with the correct apiVersions.

**Steps to reproduce the issue:**
1. Set up a cluster with Kubernetes version >1.21-0
2. Using Kustomize, attempt to render the Datadog Operator helm chart. It will fail as the chart will render the `v1beta1` components.

```yaml
---
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
helmCharts:
  - name: datadog-operator
    repo: https://helm.datadoghq.com
    version: 0.8.6
    releaseName: datadog-operator
    namespace: datadog
    includeCRDs: true
    valuesInline:
      replicaCount: 2
      apiKeyExistingSecret: datadog-secret
      appKeyExistingSecret: datadog-secret
      site: datadoghq.eu
      resources:
        limits:
          cpu: 300m
          memory: 1Gi
        requests:
          cpu: 100m
          memory: 128Mi
```

**Additional environment details (Operating System, Cloud provider, etc):**
```
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.5-gke.1504", GitCommit:"817c52cbf6d481f56d82f20f68d9bfd74f8387ef", GitTreeState:"clean", BuildDate:"2022-06-14T18:16:54Z", GoVersion:"go1.17.8b7", Compiler:"gc", Platform:"linux/amd64"}
```

### PR DESCRIPTION
**Pull request by CmdrSharp**
#### What this PR does / why we need it:
This adds the capability to override `.Capabilities.KubeVersion.GitVersion` which makes sure support for Anthos Config Management and Kustomize is maintained.

Not bumping chart version as I don't know if this warrants a new release alone.

#### Which issue this PR fixes
fixes #733 

#### Checklist
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`

### CODE REVIEW

**Review by maycmlee — APPROVED** [review]
👍 for docs

File: charts/datadog-crds/templates/datadoghq.com_datadogagents_v1beta1.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if and .Values.crds.datadogAgents (semverCompare "<=1.21-0" .Capabilities.KubeVersion.GitVersion ) }}
+{{- if and .Values.crds.datadogMetrics (semverCompare "<=1.21-0" (include "datadog-crds.kubeVersion" .)) }}
```
Comment (CharlyF) [inline charts/datadog-crds/templates/datadoghq.com_datadogagents_v1beta1.yaml:1]: s/datadogMetrics/datadogAgents

File: charts/datadog-crds/templates/datadoghq.com_datadogmonitors_v1beta1.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{{- if and .Values.crds.datadogMonitors (semverCompare "<=1.21-0" .Capabilities.KubeVersion.GitVersion ) }}
+{{- if and .Values.crds.datadogMetrics (semverCompare "<=1.21-0" (include "datadog-crds.kubeVersion" .)) }}
```
Comment (CharlyF) [inline charts/datadog-crds/templates/datadoghq.com_datadogmonitors_v1beta1.yaml:1]: s/datadogMonitors/datadogAgents

**Review by CharlyF — CHANGES_REQUESTED** [review]
Thank you for this contribution! 
Could you bump the bugfix of the chart version and add an entry to the changelog?
I added some comments, the path of the CRDs needs to match their respective files (it seems like the datadogmetrics was used for all of them).

**Comment by CmdrSharp** [comment]
@CharlyF Good catches! All fixed, and I've bumped the chart version & added a changelog entry as well.

**Comment by jackwburgess** [comment]
Hello @CharlyF @maycmlee - do we know why these checks where not successful above?

**Comment by maycmlee** [comment]
> Hello @CharlyF @maycmlee - do we know why these checks where not successful below?

I will defer to @CharlyF on that because I'm not familiar with this repo.

**Pull request closed without merging**